src/connectRoutes.js

```diff
@@ -115,8 +115,8 @@ export default (
   }
 
   const {
-    location: locationKey = 'location',
-    title: titleKey = 'title',
+    selectLocationState = state => state.location,
```

Comment:
let's keep the original external options names (`location`, `title`) so we don't need to introduce a breaking change. And let's still except strings. If a string is passed, create a selector function once right here that can be reused. A function named `selectLocation/Title` state like you already have can be used throughout the rest of the code.